path/base.go

```diff
@@ -0,0 +1,77 @@
+package path
+
+import (
+	"os"
+	"path/filepath"
+	"runtime"
+)
+
+const (
+	notation = "notation"
+)
+
+var (
+	// for mocking
+	goos          = runtime.GOOS
+	userConfigDir = os.UserConfigDir
+	userCacheDir  = os.UserCacheDir
+	getenv        = os.Getenv
+
+	// SystemLibexec directory for binaries not meant to be executed directly by users' shell or scripts
+	SystemLibexec string
+	// SystemConfig directory for configurations
+	SystemConfig string
+
+	// UserLibexec is user level libexec directory
+	UserLibexec string
+	// UserConfig is user level config directory
+	UserConfig string
+	// UserCache for user-specific cache
+	UserCache string
+)
+
+func init() {
+	loadPath()
+}
+
+// init function defines the directory for NotationLibexec, NotationConfig, NotationCache
+func loadPath() {
+	var err error
+	// set system config and libexec
+	switch goos {
+	case "darwin":
+		SystemConfig = "/Library/Application Support/notation"
+		SystemLibexec = "/usr/local/lib/notation"
+	case "windows":
+		SystemConfig = getenv("ProgramData")
+		if SystemConfig == "" {
+			panic("environment variable `ProgramData` is empty.")
```

Comment:
nit: feels like `is not set` might be better than `is empty`

Reply:
Done.